skorch/helper.py

```diff
@@ -653,6 +653,17 @@ def _initialize_optimizer(self, *args, **kwargs):
 
         return self
 
+    def train_step(self, batch, **fit_params):
+        # Call training step within the accelerator context manager
+        with self.accelerator.accumulate(self.module_):
+            # Why are we passing only module_ here, even though there might be
+            # other modules as well? First of all, there is no possibility to
+            # pass multiple modules. Second, the module_ is only used to
+            # determine if Distributed Data Parallel is being used, not for
+            # anything else. Therefore, passing module_ should be sufficient
+            # most of the time.
+            return super().train_step(batch, **fit_params)
```

Comment:
From looking at the [accelerate docs on gradient accumlation](https://huggingface.co/docs/accelerate/gradient_accumulation#the-finished-code), the scheduler needs to be prepared by the accelerator as well?

Reply:
Good catch, thanks for pointing it out. I added code to do that. Unfortunately, I don't think we can identify the usage of schedulers except if users make use of our `LRScheduler` callback. I would assume that they do, but otherwise there may be hidden bugs.